src/physical/util/labeled_graph.rs

```diff
@@ -16,23 +16,25 @@ use petgraph::{EdgeType, Graph};
 /// A [`NodeIndex`] is invalidated once a node is removed from the graph,
 /// hence the interface only permits adding new nodes.
 #[derive(Debug)]
-pub struct NodeLabeledGraph<Label, Type>
+pub struct LabeledGraph<NodeLabel, EdgeLabel, Type>
```

Comment:
Isn't this mostly or even exclusively used in the logical layer? (I'm wondering why it is defined in `physical/util`.)